hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteFunction.java

```diff
@@ -494,7 +493,7 @@ private boolean flushBucket(DataBucket bucket) {
       long waitingTime = 0L;
       long ckpTimeout = config.getLong(FlinkOptions.WRITE_COMMIT_ACK_TIMEOUT);
       long interval = 500L;
-      while (Objects.equals(instant, this.currentInstant)) {
+      while (instant == null || instant.equals(this.currentInstant)) {
```

Comment:
in the case: `instant == null`, you have returned before? 

```
    if (instant == null) {
      // in case there are empty checkpoints that has no input data
      LOG.info("No inflight instant when flushing data, skip.");
      return false;
    }
```

Reply:
Yeah, the code before is null safe, but it has problem in current master because the while loop may change the instant to be null: when the coordinator commit the last instant but not start a new one yet.